lib/shopify_app/session/session_repository.rb

```diff
@@ -5,18 +5,10 @@ class ConfigurationError < StandardError; end
     class << self
       def shop_storage=(storage)
         @shop_storage = storage
-
-        unless storage.nil? || self.shop_storage.respond_to?(:store) && self.shop_storage.respond_to?(:retrieve)
-          raise ArgumentError, "shop storage must respond to :store and :retrieve"
-        end
       end
 
       def user_storage=(storage)
         @user_storage = storage
-
-        unless storage.nil? || self.user_storage.respond_to?(:store) && self.user_storage.respond_to?(:retrieve)
-          raise ArgumentError, "user storage must respond to :store and :retrieve"
-        end
       end
 
       def retrieve_shop_session(id)
```

Comment:
Kinda related to these changes, but also not. Seems like `.user_storage` is not raising `ConfigurationError` like `.shop_storage` does. I think these two methods should be consistent.

Maybe you could squeeze in that change?

Reply:
The reason this is not done is that the app can run without the user session storage however is a lot less functional (especially in the absence of the user session) without shop storage.

Reply:
Ah so there's a case where someone would try to fetch `user_storage` even when it's not been set? I thought that might be an erroneous behavior.

Thanks for clarification!